src/redux/userReducer.js

```diff
@@ -0,0 +1,46 @@
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+
+const TOKEN =
+  "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjYxMjNmMTdlNWMzY2IxNDVjMDQ1ZTk0ZSIsInJvbGUiOiJhZG1pbiIsImlhdCI6MTYzMDk1NTk0NCwiZXhwIjoxNjMxODM0MzQ0fQ.AXc0KhpkkR9d475zpfQVH1nXugZ1SJomyQ1uJBSerRw";
+
+export const userFetch = createAsyncThunk(
+  "user/fetch",
+  async function (_, { rejectWithValue }) {
+    try {
+      const response = await fetch("http://localhost:3000/users/me", {
+        headers: { Authorization: TOKEN },
+      });
+      const responseJSON = await response.json();
+      if (!responseJSON.data) {
+        throw new Error(responseJSON.err.message);
+      }
+      return responseJSON;
+    } catch (err) {
+      return rejectWithValue(err);
+    }
+  }
+);
+
+const currentUserSlice = createSlice({
+  name: "currentUser",
+  initialState: {
+    status: "no data",
+    data: null,
+    error: null,
+  },
+  extraReducers: {
+    [userFetch.fulfilled]: (state, action) => {
```

Comment:
it's better to create a separated reducer like `saveCurrentUser()` which takes a user to put as an argument
there is no need in extraReducers right now